### PR TITLE
Github actions

### PR DESCRIPTION
**Pull request by dodget**
This PR adds automation to docker image building and pushing to Dockerhub by:

- Adding the `docker-image.yml` workflow under the `.github/workflows` directory
- Moving the relevant Dockerfile from the Dockerfiles directory to the root of the project
- Configuring the workflow so that it
  - Triggers on a push to any branch where `Dockerfile` has been modified
  - Builds the image of the modified Dockerfile
  - Logs in to Docker Hub using ATG-FAS service account credentials
  - Tags the built image with `harvardat/BRANCH_NAME:COMMIT_HASH` and `harvardat/BRANCH_NAME:latest`
  - Pushes the image to Docker Hub

## Considerations

First, in order for the workflow to work there needs to be a Dockerfile in the root of the project. Since there seems to be only one Dockerfile that is actually associated with the branch kept in the Dockerfiles directory, I figured it would be ok to simply delete that directory and move the Dockerfile to the root of the project.

Second, it means that the Dockerfile committed to the branch is no longer just for documentation

Last, it means that the branch name now carries significance in the name of the image. It may be worth considering what branch naming convention would could settle on that will also work for `harvardat` image repositories. If we settle on a useful naming convention, I'll it add to the readme for reference.

## Notes

If this PR is accepted, I'll have an identical change to the rstudio repo.

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      -
+        name: Login to Dockerhub
+        uses: docker/login-action@v1
+        with:
+          username:  ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      -
+        name: Set environment variables for image tagging
+        run: echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
+        run: echo "BRANCH_NAME=`echo ${GITHUB_REF##*/}`" >> $GITHUB_ENV
```
Comment (arthurian) [inline .github/workflows/docker-image.yml]: _Question about the yaml syntax_: doesn't the duplicate `run:` key clobber the previous one when the YAML is parsed? I think you might have to do it as a multi-line string as shown in one of the [workflow examples](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun):

```yaml
run: |
  echo "SHORT_SHA=`echo ${GITHUB_SHA} | cut -c1-8`" >> $GITHUB_ENV
  echo "BRANCH_NAME=`echo ${GITHUB_REF##*/}`" >> $GITHUB_ENV
```

_Suggestion for BRANCH_NAME:_ I noticed that if the branch name contains a slash (e.g. `cs109a/medium`), we end up with just the last part after the final slash (e.g. `medium`).  What if we simply chopped the prefix (e.g. `refs/heads/`) and replaced any slashes, so in the above example we'd get `cs109a-medium`. There's probably a more elegant way to do it, but something like this maybe?

```sh
BRANCH_NAME=${GITHUB_REF/refs\/heads\//}
BRANCH_NAME=${BRANCH_NAME/\//-}
```

**Review by arthurian — APPROVED** [review]
@dodget This is a slick workflow. Having a single Dockerfile in the root and pushing the image to a Dockerhub repository named after the branch should work fine for our use cases.

Since we want to use this workflow for our [jupyter](https://github.com/fasrc/fas-ondemand-jupyter-app) and [rstudio](https://github.com/fasrc/fas-ondemand-rstudio) apps (possibly others in the future), I'd like to propose that in addition to using the branch name for the repository name, we also prefix the branch name with the application type (e.g. jupyter or rstudio). 

This would ensure that two branches of the same name are pushed to separate Dockerhub repositories and that they are labeled according to the kind of image they contain. At the very least, this would avoid a single `master` repository with both jupyter and rstudio images. On that point, this might also be a good opportunity to [rename the default branch](https://github.com/github/renaming).

**Comment by dodget** [comment]
@arthurian - Yes, I totally agree about the prefixing the application type in the image tag. I had originally included that, but removed it for no good reason. Thanks for the other suggestions; I'll make those updates!

**Review by joshuagetega — APPROVED** [review]
Looks good, Tylor!

**Comment by pontiggi** [comment]
LGTM

**Review by pontiggi — APPROVED** [review]
LGTM

**Pull request merged**